### PR TITLE
Protecting rounding issues and bin size issues in BField plotting

### PR DESCRIPTION
**Pull request by michaelmackenzie**
Adding rounding when making bin numbers, and checking that the number of bins is consistent with the map grid size.

Also adding a small step into the map when at the edge of the histogram, 1% of the map grid size.

### CODE REVIEW

File: Analyses/src/BFieldPlotter_module.cc [hunk 1]
```diff
@@ -79,15 +80,15 @@ namespace mu2e {
     if(_axisOneMin >= _axisOneMax || _axisTwoMin >= _axisTwoMax) {
       throw cet::exception("BADCONFIG") << "BField mapping plane ill defined: "
 					<< _axisOneMin << " < axisOneValues < " << _axisOneMax << ", "
-					<< _axisTwoMin << " < axisTwoValues < " << _axisTwoMax;
+					<< _axisTwoMin << " < axisTwoValues < " << _axisTwoMax << endl;
     }
```
Comment (kutschke) [inline Analyses/src/BFieldPlotter_module.cc]: The exception system strips std::endl.  If you want to force a break use "\n".

**Comment by kutschke** [comment]
@FNALbuild run build test

File: Analyses/src/BFieldPlotter_module.cc (hunk 1 above)
Comment (michaelmackenzie) [reply Analyses/src/BFieldPlotter_module.cc]: Good to know, I'll simply let it not end the line as it originally was in that case.

**Comment by kutschke** [comment]
@FNALbuild run build test

**Pull request merged**